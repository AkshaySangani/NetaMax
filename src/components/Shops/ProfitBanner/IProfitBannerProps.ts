export interface IProfitBannerProps {
  /**
   * Shop's current profit
   * @type {number}
   */
  todaysProfit: number;

  /**
   * The share url
   * @type {string}
   */
  url: string;
}
