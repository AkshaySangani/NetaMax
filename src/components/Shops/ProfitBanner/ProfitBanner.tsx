import React, { ReactElement } from "react";

import { CURRENCY_NAME } from "constants/localeFormatConstants";
import Link from "next/link";
import { formatMoney } from "utils/currencyUtils";

import { Center, Stack, Text } from "@chakra-ui/react";

import { IProfitBannerProps } from "./IProfitBannerProps";

/**
 * The ProfitBanner component.
 * @returns {ReactElement} the ProfitBanner component.
 * @param {IProfitBannerProps} profitBannerProps ProfitBanner props
 */
export const ProfitBanner = (profitBannerProps: IProfitBannerProps): ReactElement => {
  const { todaysProfit, url } = profitBannerProps;

  return (
    <div>
      <Link href={url} passHref>
        <a href="inherit" target="_blank">
          <Center
            mt="2"
            cursor="pointer"
            background={"linear-gradient(to right, #d64900 0%, #F52220 100%)"}
            h={["70px", "100px", "115px"]}
            color="white"
            borderRadius="md"
            mb="4"
          >
            <Stack spacing={2} align={"center"} lineHeight={["14px", "20px", "24px"]}>
              <Text
                fontSize={{ base: "15px", sm: "20px", md: "20px", lg: "20px" }}
                fontWeight="500"
              >
                Tus ganancias hasta hoy son de{" "}
                <b>¡{formatMoney(todaysProfit, CURRENCY_NAME)} pesos!</b>
              </Text>
              <Text
                fontSize={{ base: "18px", sm: "23px", md: "24px", lg: "24px" }}
                fontWeight="700"
              >
                ¡Comparte tu liga y sigue ganando!
              </Text>
            </Stack>
          </Center>
        </a>
      </Link>
    </div>
  );
};
