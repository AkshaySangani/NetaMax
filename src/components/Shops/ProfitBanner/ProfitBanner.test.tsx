import { render, cleanup, screen } from "@testing-library/react";
import { ProfitBanner } from "./ProfitBanner";
import { IProfitBannerProps } from "./IProfitBannerProps";

describe("ProfitBanner Component", () => {
  let mock: IProfitBannerProps;

  beforeEach(() => {
    mock = {
      todaysProfit: 1000,
      url: "https://neta.mx/",
    };
  });

  afterEach(() => {
    cleanup();
  });

  test("ProfitBanner should be render correctly", () => {
    const { container } = render(<ProfitBanner {...mock} />);
    expect(container).toMatchSnapshot();
  });

  test("Link should be render properly", () => {
    render(<ProfitBanner {...mock} />);
    expect(screen.getByRole("link")).toHaveAttribute("href", mock.url);
  });
});
