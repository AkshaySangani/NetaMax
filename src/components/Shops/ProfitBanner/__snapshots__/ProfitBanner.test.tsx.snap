// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfitBanner Component ProfitBanner should be render correctly 1`] = `
<div>
  <div>
    <a
      href="https://neta.mx/"
      target="_blank"
    >
      <div
        class="css-ky3xz8"
      >
        <div
          class="chakra-stack css-1moilrc"
        >
          <p
            class="chakra-text css-lxkbi3"
          >
            Tus ganancias hasta hoy son de
             
            <b>
              ¡
              $1,000.00
               pesos!
            </b>
          </p>
          <p
            class="chakra-text css-xatlyb"
          >
            ¡Comparte tu liga y sigue ganando!
          </p>
        </div>
      </div>
    </a>
  </div>
</div>
`;
